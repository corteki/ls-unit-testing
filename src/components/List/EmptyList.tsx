import React, { FC } from 'react';
import { EmptyListProps } from './EmptyListProps';
import './List.scss';

export const EmptyList: FC<EmptyListProps> = ({message}) => (
  <div className="List">
    <option className="List__option">{message}</option>
  </div>
);
