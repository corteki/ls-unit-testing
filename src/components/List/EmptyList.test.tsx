import React from 'react';
import renderer from 'react-test-renderer';
import { EmptyList } from './EmptyList';

describe('<EmptyList/>', () => {
  it('should render a single option that displays the message "No results"', () => {
    const tree = renderer
    .create(
      <EmptyList message="No results"/>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
