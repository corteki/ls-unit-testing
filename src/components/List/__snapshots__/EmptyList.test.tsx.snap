// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyList/> should render a single option that displays the message "No results" 1`] = `
<div
  className="List"
>
  <option
    className="List__option"
  >
    No results
  </option>
</div>
`;
